imap/index.c

```diff
@@ -6132,6 +6133,32 @@ MsgData **index_msgdata_load(struct index_state *state,
     if (found_anchor)
         *found_anchor = 0;
 
+    /* set mailbox level states */
+    for (j = 0; sortcrit[j].key; j++); // count how many we need
+    preload = xzmalloc(j * sizeof(int));
```

Comment:
j can be zero, which should be fine with malloc(0) returning either NULL or a free-able pointer. But I remember we had issues with freeing a zero size malloc result (and fixed that during a late evening hack session at a bar in Zurich :))

Reply:
Good point, fixed in the updated commit.